plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -196,7 +196,11 @@ public ConnectorPageSource createPageSource(
                 dataSourceStats,
                 options.withSmallFileThreshold(getParquetSmallFileThreshold(session))
                         .withVectorizedDecodingEnabled(isParquetVectorizedDecodingEnabled(session)),
-                timeZone, dynamicFilter, enablePredicatePushDown);
+                timeZone, dynamicFilter, isBaseFileOnly);
```

Comment:
We need to handle the schema differently if we directly use the parquet page source, i.e., removing the meta columns if needed.

Reply:
Sg.  In the future, the `Page`-based Trino reader context should hide such logic of handling schema and returning page source of the base file only, so the current shape is OK.